tensorflow/core/kernels/dense_to_sparse_batch_dataset_op.cc

```diff
@@ -49,10 +49,7 @@ class DenseToSparseBatchDatasetOp : public UnaryDatasetOpKernel {
     OP_REQUIRES_OK(ctx, ctx->input("row_shape", &row_shape_t));
     OP_REQUIRES(ctx, TensorShapeUtils::IsVector(row_shape_t->shape()),
                 errors::InvalidArgument("row_shape must be a vector"));
-    TensorShape row_shape;
-    for (size_t i = 0; i < row_shape_t->dim_size(0); ++i) {
-      row_shape.AddDim(row_shape_t->vec<int64>()(i));
-    }
+    PartialTensorShape row_shape(row_shape_t->vec<int64>());
```

Comment:
This could potentially lead to an error (e.g. if someone smuggled a `-2` into the `row_shape` tensor via a placeholder). Could you replace this with something like:

```c++
PartialTensorShape row_shape;
OP_REQUIRES_OK(ctx, PartialTensorShape::MakePartialShape(row_shape_t->vec<int64>().data(), row_shape_t->NumElements(), &row_shape));
```

...and add a test where the `row_shape` gets an invalid value (e.g. via a placeholder)?

Reply:
Sure, good suggestion for me!

Reply:
Done, @mrry . Could you take a look again?